### PR TITLE
- add Unit Test for method findCarEntityByModel(String model) from CarRepository 

### PR DESCRIPTION
**Pull request by Miller1995**
- add new dependency in pom.xml for H2 database;
 - add new file application.properties in package test.resources;
- add new conf for H2 database in memory for testing;
- add new unit test "CarRepositoryTest" for method findByModel from carRepositories;
- in this test class add annotation @DataJpaTest for using H2 database in memory, used dependency injection, generate method setup() with @BeforeEach that create an object carEntity and saved this object in mem database before run test, after generate tearDown() method with @AfterEach which assign null for carEntity object and after delete all data from DB;
- add new 2 test ( positive and negative case ) and build the logic for these tests;

### CODE REVIEW

File: src/test/java/md/miller1995/Dealership/repositories/CarRepositoryTest.java [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.orm.jpa.DataJpaTest;
+
+import java.time.LocalDate;
+import java.util.List;
+
+import static org.assertj.core.api.Assertions.assertThat;
```
Comment (AlexandruIliescu) [inline src/test/java/md/miller1995/Dealership/repositories/CarRepositoryTest.java:13]: Please keep an empty row between the last import and the class or the annotations of the class.

File: src/test/java/md/miller1995/Dealership/repositories/CarRepositoryTest.java [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+@DataJpaTest
+class CarRepositoryTest {
+
+    @Autowired
+    private CarRepository carRepository;
+    CarEntity carEntity;
```
Comment (AlexandruIliescu) [inline src/test/java/md/miller1995/Dealership/repositories/CarRepositoryTest.java:19]: can `CarEntity` be also private?

File: src/test/java/md/miller1995/Dealership/repositories/CarRepositoryTest.java [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+
+    @BeforeEach
+    void setUp() {
+
```
Comment (AlexandruIliescu) [inline src/test/java/md/miller1995/Dealership/repositories/CarRepositoryTest.java:23]: There are too many empty rows in this class.

File: src/test/java/md/miller1995/Dealership/repositories/CarRepositoryTest.java [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+                LocalDate.now(),
+                "blue",
+                2999
+        );
+
+        carRepository.save(carEntity);
+    }
+
+    @AfterEach
+    void tearDown() {
```
Comment (AlexandruIliescu) [inline src/test/java/md/miller1995/Dealership/repositories/CarRepositoryTest.java:39]: When we are working with the repository layer, we want to create needed resources for each test and have a teardown method after all the test methods were executed (not after each test). In this way, we can save some resources.

Let's consider 2 examples, as a flow:
First approach:
```
1. Create a CarEntity before each test
2. Run test 1
3. Erase all the database records after each test

1. Create a CarEntity before each test
2. Run test 2
3. Erase all the database records after each test
```

Second approach:
```
1. Create a CarEntity inside the test in such a way you need it (maybe with some custom fields)
2. Run test 1

1. Create a CarEntity inside the test in such a way you need it (maybe with some custom fields)
2. Run test 2
...
3. Erase all the database records after all the tests were executed
```

**Review by AlexandruIliescu — APPROVED** [review]

File: src/test/java/md/miller1995/Dealership/repositories/CarRepositoryTest.java (hunk 1 above)
Comment (Miller1995) [reply src/test/java/md/miller1995/Dealership/repositories/CarRepositoryTest.java:13]: - add empty row  between the last import and the annotation

File: src/test/java/md/miller1995/Dealership/repositories/CarRepositoryTest.java (hunk 2 above)
Comment (Miller1995) [reply src/test/java/md/miller1995/Dealership/repositories/CarRepositoryTest.java:19]: - changed access modifier for carentity field

File: src/test/java/md/miller1995/Dealership/repositories/CarRepositoryTest.java (hunk 3 above)
Comment (Miller1995) [reply src/test/java/md/miller1995/Dealership/repositories/CarRepositoryTest.java:23]: - deleted empty rows

File: src/test/java/md/miller1995/Dealership/repositories/CarRepositoryTest.java (hunk 4 above)
Comment (Miller1995) [reply src/test/java/md/miller1995/Dealership/repositories/CarRepositoryTest.java:39]: understood, but until I have only 2 tests for repository layer, I will use the first approach and after that I will update my code
